packages/labs/eleventy-plugin-lit/src/test/eleventy-plugin-test.ts

```diff
@@ -171,370 +178,340 @@ test('without plugin', async ({rig}) => {
-        render() {
-          return html\`<slot></slot>\`;
-        }
-      }
-      customElements.define('my-container', MyContainer);
-
-      class MyContent extends LitElement {
+      class MyElement extends LitElement {
         render() {
           return html\`<b>shadow content</b>\`;
         }
       }
-      customElements.define('my-content', MyContent);
-    `,
-  });
-  assert.equal(
-    (
-      await rig.exec(
-        'NODE_OPTIONS=--experimental-vm-modules eleventy --config .eleventy.cjs'
-      ).done
-    ).code,
-    0
-  );
-  assert.equal(
-    await rig.read('_site/index.html'),
-    normalize(`
-      <h1>Heading 1</h1>
-      <my-container><template shadowroot="open"><!--lit-part Pz0gobCCM4E=--><slot></slot><!--/lit-part--></template>
-        <my-content><template shadowroot="open"><!--lit-part 8dHorjH6jDo=--><b>shadow content</b><!--/lit-part--></template></my-content>
-      </my-container>
-      <h1>Heading 2</h1>
-      <p><my-container><template shadowroot="open"><!--lit-part Pz0gobCCM4E=--><slot></slot><!--/lit-part--></template></my-container></p>
-    `)
-  );
-});
-
-test('missing component definition', async ({rig}) => {
-  await rig.write({
-    // md
-    'index.md': `
-      # Heading
-      <my-element></my-element>
-    `,
-
-    // config
-    '.eleventy.cjs': `
-      const litPlugin = require('@lit-labs/eleventy-plugin');
-      module.exports = function (eleventyConfig) {
-        eleventyConfig.addPlugin(litPlugin, {
-          componentModules: [],
-        });
-      };
+      customElements.define('my-element', MyElement);
     `,
-  });
-  assert.equal(
-    (
-      await rig.exec(
-        'NODE_OPTIONS=--experimental-vm-modules eleventy --config .eleventy.cjs'
-      ).done
-    ).code,
-    0
-  );
-  // TODO(aomarks) There should be an error message about missing components.
-  assert.equal(
-    await rig.read('_site/index.html'),
-    normalize(`
-      <h1>Heading</h1>
-      <p><my-element></my-element></p>
-    `)
-  );
-});
 
-test.skip('watch mode', async ({rig}) => {
-  await rig.write({
     // eleventy config
     '.eleventy.cjs': `
       const litPlugin = require('@lit-labs/eleventy-plugin');
       module.exports = function (eleventyConfig) {
         eleventyConfig.addPlugin(litPlugin, {
           componentModules: ['./js/my-element.js'],
+          mode: '${mode}'
         });
-        eleventyConfig.addWatchTarget('js/my-element.js');
       };
     `,
+  };
 
-    // markdown
-    'index.md': `
-      # Heading
-      <my-element></my-element>
-    `,
+  const baseCommandToExec =
+    (mode === 'vm' ? 'NODE_OPTIONS=--experimental-vm-modules ' : '') +
+    'eleventy --config .eleventy.cjs';
 
-    // initial component definition
-    'js/my-element.js': `
-      import { html, LitElement } from 'lit';
-      class MyElement extends LitElement {
-        render() {
-          return html\`INITIAL\`;
+  test('basic component in markdown file', async ({rig}) => {
```

Comment:
Sorry I didn't think of this before -- Could you add a new test to confirm that we can process multiple pages simultaneously, which would cover the `id` tracking feature? Maybe the pages could be generated so that you can easily make a large number of them (e.g. 25 pages, each with an incrementing number in it).

(Sorry if there is one, but I don't think I see it).

Reply:
Oh, yes. Good call out.